hive/src/main/java/org/apache/zeppelin/hive/HiveInterpreter.java

```diff
@@ -0,0 +1,203 @@
+
+import java.sql.*;
+import java.util.List;
+import java.util.Properties;
+
+import com.nflabs.zeppelin.interpreter.*;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.nflabs.zeppelin.interpreter.InterpreterResult.Code;
+import com.nflabs.zeppelin.scheduler.Scheduler;
+import com.nflabs.zeppelin.scheduler.SchedulerFactory;
+
+/**
+ * Hive interpreter for Zeppelin.
+ */
+public class HiveInterpreter extends Interpreter {
+  Logger logger = LoggerFactory.getLogger(HiveInterpreter.class);
+  int commandTimeOut = 600000;
+
+  static final String HIVESERVER_URL = "hive.hiveserver2.url";
+  static final String HIVESERVER_USER = "hive.hiveserver2.user";
+  static final String HIVESERVER_PASSWORD = "hive.hiveserver2.password";
+
+  static {
+    Interpreter.register(
+      "hive",
+      "hive",
+      HiveInterpreter.class.getName(),
+      new InterpreterPropertyBuilder()
+        .add(HIVESERVER_URL, "jdbc:hive2://localhost:10000", "The URL for HiveServer2.")
+        .add(HIVESERVER_USER, "hive", "The hive user")
+        .add(HIVESERVER_PASSWORD, "", "The password for the hive user").build());
+  }
+
+  public HiveInterpreter(Properties property) {
+    super(property);
+  }
+
+  Connection jdbcConnection;
+  Exception exceptionOnConnect;
+
+  //Test only method
+  public Connection getJdbcConnection()
+      throws SQLException {
+    String url = getProperty(HIVESERVER_URL);
+    String user = getProperty(HIVESERVER_USER);
+    String password = getProperty(HIVESERVER_PASSWORD);
+
+    return DriverManager.getConnection(url, user, password);
+  }
+
+  @Override
+  public void open() {
+    logger.info("Jdbc open connection called!");
+    try {
+      String driverName = "org.apache.hive.jdbc.HiveDriver";
+      Class.forName(driverName);
+    } catch (ClassNotFoundException e) {
+      logger.error("Can not open connection", e);
+      exceptionOnConnect = e;
+      return;
+    }
+    try {
+      jdbcConnection = getJdbcConnection();
+      exceptionOnConnect = null;
+      logger.info("Successfully created Jdbc connection");
+    }
+    catch (SQLException e) {
+      logger.error("Cannot open connection", e);
+      exceptionOnConnect = e;
+    }
+  }
+
+  @Override
+  public void close() {
+    try {
+      if (jdbcConnection != null) {
+        jdbcConnection.close();
+      }
+    }
+    catch (SQLException e) {
+      logger.error("Cannot close connection", e);
+    }
+    finally {
+      jdbcConnection = null;
+      exceptionOnConnect = null;
+    }
+  }
+
+  Statement currentStatement;
+  private InterpreterResult executeSql(String sql) {
+    try {
+      if (exceptionOnConnect != null) {
+        return new InterpreterResult(Code.ERROR, exceptionOnConnect.getMessage());
+      }
+      currentStatement = jdbcConnection.createStatement();
+      StringBuilder msg = null;
+      if (StringUtils.containsIgnoreCase(sql, "EXPLAIN ")) {
```

Comment:
The result is a rowset with one column, indistinguishable from a query that returns a single column.